docs/03_server/09_data-pipelines/03_implementations.mdx

```diff
@@ -0,0 +1,185 @@
+
+The type of object that you would like this source to return must be given as part of the source definition, example shown below.
+
+When configuring the db batch poller to use as part of your pipeline you must provide the following as part of the configuration:
+- source (required): this is what the PersistenceManager will use to find the last persisted record and should be should be unique across all pipelines
+- index (required): this is the table index used to order the records and get the value for the last persisted record for recovery - if single field index is provided, ensure this value is of type Long. Else, see below.
+- buildIndex (required for multi field index only): if you would like to use an index which contains multiple fields, you need to provide a lambda which converts from a Long value (the last persisted value) to your EntityIndex object using this value - you must ensure that the Long value which you would like to be used as the persistence index needs to be the final value inside your index object
+- dbLookupDelayMs (optional): this is the delay time that the poller will wait for in milliseconds after not receiving any new data from the database - it is configured to 200ms by default and is not required as part of the provided config
+
+GPAL Example:
+```kotlin
+// Single Field Index Example
+val batchPollSource = dbBatchQuery<TestIndexTable> {
+    source = "SINGLE_INDEX_SOURCE"
+    index = TestIndexTable.ByIndex
+}
+
+// Multiple Field Index Example
+val batchPollMultiIndexSource = dbBatchQuery<TestMultiIndexTable> {
+    source = "MULTI_INDEX_SOURCE"
+    index = TestMultiIndexTable.ByNameIndex
+    buildIndex { value -> TestMultiIndexTable.ByNameIndex("NAME", value) }
+}
+
+pipelines {
+    pipeline("TEST_BATCH_POLL_PIPELINE") {
+        source(batchPollSource)
+            .sink(testSink)
+    }
+}
+```
+
+## Operators
+
+### CSV Decoder
+
+Split Operator to decode from an InputStream (from a PipelineFile object) to a CsvRow object which contains:
+
+1. the line number for each CSV row
+2. an object of a user defined type representing the data of each CSV line, by default this is of type `Map<String, String>`
+
+GPAL functions available are `csvRawDecoder()` to deserialise to `Map<String, String>` and `csvDecoder<Trade>()` to deserialise to specified class.
+
+GPAL Examples:
+```kotlin
+// Decoding to Map<String, String>
+pipeline("TEST_PIPELINE_RAW_DECODER") {
+    source(camelSource {
+        location = ""
```

Comment:
```suggestion
        location = "file://src/test/resources/camel-source-test?noop=true&idempotent=true"
```

Reply:
Just to be clear that an empty string is not a valid example (may not be required to point out but you never know!)